### PR TITLE
[Streamango] Add new extractor

### PR DESCRIPTION
**Pull request by mphe**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Add extractor for http://streamango.com/

### CODE REVIEW

**Comment by dstftw** [comment]
All formats should be extracted.

**Comment by mphe** [comment]
Done, added dash support.
But there's a problem. For some reason the download gets stuck when using dash. Sometimes, after a while, it continues just to get stuck again immediately. I have no idea what's causing this, maybe server side problems.
Is there a way to set the normal mp4 download as default?

**Comment by dstftw** [comment]
>For some reason the download gets stuck when using dash

Works fine for me.

File: youtube_dl/extractor/streamango.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        }
+    }, {
+        'url': 'https://streamango.com/embed/clapasobsptpkdfe/20170315_150006_mp4',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        def extract_url(urltype):
+            return self._search_regex(
+                r'type\s*:\s*["\']{}["\']\s*,\s*src\s*:\s*["\'](?P<url>.+?)["\'].*'.format(urltype),
```
Comment (dstftw) [inline youtube_dl/extractor/streamango.py]: `{}` does not work in python 2.6.

File: youtube_dl/extractor/streamango.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                webpage, 'video URL', group='url')
+
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._og_search_title(webpage)
+        url = 'https:' + extract_url('video/mp4')
+        dashurl = extract_url(r'application/dash\+xml')
```
Comment (dstftw) [inline youtube_dl/extractor/streamango.py]: Extraction should not break if one of the formats is missing.

File: youtube_dl/extractor/streamango.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        formats = [{
+            'url': url,
+            'ext': 'mp4',
+            'width': 1280,
+            'height': 720,
```
Comment (dstftw) [inline youtube_dl/extractor/streamango.py]: This is not true in general.

File: youtube_dl/extractor/streamango.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class StreamangoIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?streamango\.com/(?:f|embed)/(?P<id>.+?)/(?:.+)'
```
Comment (dstftw) [inline youtube_dl/extractor/streamango.py]: `https://streamango.com/f/clapasobsptpkdfe` is not matched by this regex.

**Comment by dstftw** [comment]
Do you want to be credited as one of the authors of youtube-dl? If so under what name?

**Pull request closed without merging**